OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -178,10 +178,30 @@ static class DMOpcodeHandlers {
                             formattedString.Append(value.Stringify());
                             break;
                         }
-                        case StringFormatTypes.Ref: {
-                            DreamObject refObject = state.Pop().GetValueAsDreamObject();
-
-                            formattedString.Append(refObject.CreateReferenceID(state.DreamManager));
+                        case StringFormatTypes.Ref:
```

Comment:
This should be moved out into shared code that both this and the `ref()` proc uses